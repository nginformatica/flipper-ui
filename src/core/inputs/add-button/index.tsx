import React from 'react'
import type { IButtonProps } from '@/core/inputs/button'
import { IconNoteAdd } from '@/icons/mui'
import { ButtonAdd } from './styles'

export interface Props extends IButtonProps {
    label?: string
    name: string
    margin?: string | number
    padding?: string | number
    disabled?: boolean
    fullWidth?: boolean
    onClick?(): void
}

const AddButton = (props: Props) => (
    <ButtonAdd
        {...props}
        dashed
        color='primary'
        id={'id-add-' + props.name}
        name={'add-' + props.name}
        padding={props.padding}
        disabled={!!props.disabled}
        margin={props.margin || '12px 0 24px'}
        fullWidth={props.fullWidth}
        onClick={props.onClick}>
        <IconNoteAdd />
        {props.label}
    </ButtonAdd>
)

export default AddButton
