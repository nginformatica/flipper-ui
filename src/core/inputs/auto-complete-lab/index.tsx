import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { default as MuiAutocomplete } from '@material-ui/lab/Autocomplete'
export type {
    AutocompleteRenderInputParams,
    AutocompleteProps,
    AutocompleteCloseReason,
    AutocompleteChangeReason,
    AutocompleteInputChangeReason,
    AutocompleteGetTagProps,
    AutocompleteRenderOptionState
} from '@material-ui/lab/Autocomplete'

const useStyles = makeStyles({
    inputRoot: {
        '&.MuiOutlinedInput-root': {
            padding: '0 62px 0 8px'
        }
    }
})

export const AutoComplete: typeof MuiAutocomplete = props => {
    const { classes, ...otherProps } = props
    const styles = useStyles(props)

    return (
        <MuiAutocomplete
            {...otherProps}
            classes={{
                inputRoot: styles.inputRoot,
                ...classes
            }}
        />
    )
}
