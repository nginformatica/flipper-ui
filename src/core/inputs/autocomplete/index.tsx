import React from 'react'
import MuiAutocomplete from '@mui/material/Autocomplete'
export type {
    AutocompleteChangeReason,
    AutocompleteCloseReason,
    AutocompleteInputChangeReason,
    AutocompleteProps,
    AutocompleteRenderGetTagProps,
    AutocompleteRenderInputParams,
    AutocompleteRenderOptionState
} from '@mui/material/Autocomplete'

export const Autocomplete: typeof MuiAutocomplete = props => {
    const { ...otherProps } = props

    return <MuiAutocomplete {...otherProps} />
}
