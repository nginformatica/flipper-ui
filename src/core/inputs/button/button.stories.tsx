import React from 'react'
import type { Meta, StoryFn } from '@storybook/react'
import Button from '.'
import { Add as IconAdd } from '@/icons'

export default {
    title: 'Inputs/Button',
    component: Button,
    argTypes: {
        backgroundColor: { control: 'color' }
    }
} as Meta<typeof Button>

const Template: StoryFn<typeof Button> = args => <Button {...args} />

const commonArgs = {
    margin: 12,
    children: 'Button'
}

export const Default = Template.bind({})
Default.args = {
    ...commonArgs
}

export const Primary = Template.bind({})
Primary.args = {
    ...commonArgs,
    color: 'primary'
}

export const Secondary = Template.bind({})
Secondary.args = {
    ...commonArgs,
    color: 'secondary'
}

export const ContainedPrimary = Template.bind({})
ContainedPrimary.args = {
    ...commonArgs,
    variant: 'contained',
    color: 'primary'
}

export const ContainedSecondary = Template.bind({})
ContainedSecondary.args = {
    ...commonArgs,
    variant: 'contained',
    color: 'secondary'
}

export const OutlinedPrimary = Template.bind({})
OutlinedPrimary.args = {
    ...commonArgs,
    variant: 'outlined',
    color: 'primary'
}

export const OutlinedSecondary = Template.bind({})
OutlinedSecondary.args = {
    ...commonArgs,
    variant: 'outlined',
    color: 'secondary'
}

export const Selected = Template.bind({})
Selected.args = {
    ...commonArgs,
    variant: 'contained',
    color: 'primary',
    selected: true
}

export const DashedPrimary = Template.bind({})
DashedPrimary.args = {
    ...commonArgs,
    variant: 'dashed',
    color: 'primary'
}

export const DashedSecondary = Template.bind({})
DashedSecondary.args = {
    ...commonArgs,
    variant: 'dashed',
    color: 'secondary'
}

export const AddIcon = Template.bind({})
AddIcon.args = {
    variant: 'fab',
    color: 'primary',
    margin: 12,
    children: <IconAdd />
}

export const Small = Template.bind({})
Small.args = {
    ...commonArgs,
    size: 'small',
    variant: 'contained',
    color: 'primary'
}

export const Medium = Template.bind({})
Medium.args = {
    ...commonArgs,
    size: 'medium',
    variant: 'contained',
    color: 'primary'
}

export const Large = Template.bind({})
Large.args = {
    ...commonArgs,
    size: 'large',
    variant: 'contained',
    color: 'primary'
}

export const Disabled = Template.bind({})
Disabled.args = {
    ...commonArgs,
    variant: 'contained',
    color: 'primary',
    disabled: true
}
