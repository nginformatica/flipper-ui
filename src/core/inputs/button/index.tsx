import { Button as MuiButton } from '@material-ui/core'
import React, { ElementType, MouseEvent } from 'react'
import { DefaultProps } from '../../types'

export interface ButtonProps extends DefaultProps {
    disabled?: boolean
    selected?: boolean
    disableTouchRipple?: boolean
    component?: ElementType
    color?: 'default' | 'primary' | 'inherit' | 'secondary'
    size?: 'small' | 'medium' | 'large'
    href?: string
    fullWidth?: boolean
    variant?: 'text' | 'outlined' | 'contained' | 'dashed'
    target?: string
    children?: React.ReactNode
    'data-testid'?: string
    onClick?(event: MouseEvent<HTMLButtonElement>): void
}

const Button = ({
    children,
    margin,
    padding,
    style = {},
    variant,
    ...otherProps
}: ButtonProps) => (
    <MuiButton
        {...otherProps}
        variant={variant === 'dashed' ? 'outlined' : variant}
        style={{
            ...style,
            margin,
            padding,
            opacity: otherProps.selected ? 0.5 : 1,
            borderStyle: variant === 'dashed' ? 'dashed' : 'initial'
        }}>
        {children}
    </MuiButton>
)

export default Button
