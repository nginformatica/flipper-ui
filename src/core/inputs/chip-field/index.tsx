import React, { useMemo } from 'react'
import type { ReactNode } from 'react'
import ChipInput from 'material-ui-chip-input'
import { styled } from 'styled-components'
import type { InputLabelProps } from '@material-ui/core'
import type { ChipRenderer } from 'material-ui-chip-input'

export interface ChipFieldProps<T extends TChipValues> {
    allowDuplicates?: boolean
    values: T[]
    variant?: 'outlined' | 'standard' | 'filled'
    disabled?: boolean
    fullWidth?: boolean
    fullWidthInput?: boolean
    InputLabelProps?: InputLabelProps
    label?: ReactNode
    placeholder?: string
    readOnly?: boolean
    error?: boolean
    chipRenderer?: ChipRenderer
    onAdd: (values: string) => void
    onDelete: (value: string, index?: number) => void
}

export type TChipValues = {
    value: string | number
}

const StyledChipInput = styled(ChipInput)`
    & > div > div {
        padding: 0px !important;
    }
    & > div > div > div {
        margin: 2px;
    }
    & > label {
        top: -6px !important;
        font-size: 14px !important;
    }
    & > label.MuiInputLabel-shrink {
        top: 0px !important;
    }
    & > div > div > input {
        padding: 10px !important;
        font-size: 14px !important;
    }
`

export const ChipField = <T extends TChipValues>(props: ChipFieldProps<T>) => {
    const values = useMemo(
        () => props.values.map(item => item.value),
        [props.values]
    )

    return (
        <StyledChipInput
            {...props}
            alwaysShowPlaceholder
            id='chip-field-root'
            value={values}
        />
    )
}

export default ChipField
