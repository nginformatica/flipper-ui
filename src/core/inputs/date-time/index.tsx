import React from 'react'
import type { MouseEvent } from 'react'
import DateFnsUtils from '@date-io/date-fns'
import {
    KeyboardTimePicker,
    KeyboardDatePicker,
    KeyboardDateTimePicker,
    MuiPickersUtilsProvider
} from '@material-ui/pickers'
import type { DefaultProps } from '../../types'
import type { KeyboardDatePickerProps } from '@material-ui/pickers/DatePicker'
import type { KeyboardDateTimePickerProps } from '@material-ui/pickers/DateTimePicker'
import type { KeyboardTimePickerProps } from '@material-ui/pickers/TimePicker'
import type { MaterialUiPickersDate } from '@material-ui/pickers/typings/date'

export interface IProps {
    locale?: DateFnsUtils['locale']
    type?: 'date' | 'time' | 'datetime'
    inputProps?: object
    onAuxClick?(event: MouseEvent<HTMLDivElement, MouseEvent>): void
    onAuxClickCapture?(event: MouseEvent<HTMLDivElement, MouseEvent>): void
    onChange(date: MaterialUiPickersDate | null, value?: string): void
}

const DEFAULT_FORMATS = {
    date: 'dd/MM/yyyy',
    time: 'HH:mm',
    datetime: 'dd/MM/yyyy HH:mm'
}

export type DateTimeProps = Omit<
    KeyboardDatePickerProps,
    'margin' | 'onChange'
> &
    Omit<KeyboardDateTimePickerProps, 'margin' | 'onChange'> &
    Omit<KeyboardTimePickerProps, 'margin' | 'onChange'> &
    IProps &
    DefaultProps

const DateTime = ({
    padding,
    margin,
    style,
    format,
    variant = 'inline',
    inputVariant = 'outlined',
    ampm = false,
    invalidDateMessage = '',
    minDateMessage = '',
    maxDateMessage = '',
    locale,
    type = 'date',
    ...otherProps
}: DateTimeProps) => {
    const fieldProps = {
        ...otherProps,
        format: format ? format : DEFAULT_FORMATS[type],
        variant,
        inputVariant,
        ampm,
        invalidDateMessage,
        minDateMessage,
        maxDateMessage,
        style: { margin, padding, ...style },
        inputProps: {
            autoComplete: 'off',
            role: 'date-picker',
            ...otherProps.inputProps
        },
        InputAdornmentProps: {
            style: { width: '32px' },
            ...otherProps.InputAdornmentProps
        },
        InputLabelProps: {
            ...otherProps.InputLabelProps
        },
        InputProps: {
            style: { fontSize: '14px' },
            ...otherProps.InputProps
        }
    }

    const renderDatePicker = () => (
        <KeyboardDatePicker {...fieldProps} size='small' />
    )

    const renderTimePicker = () => (
        <KeyboardTimePicker {...fieldProps} size='small' />
    )

    const renderDateTimePicker = () => (
        <KeyboardDateTimePicker {...fieldProps} size='small' />
    )

    const renderChildren = () => {
        if (type === 'date') {
            return renderDatePicker()
        }

        if (type === 'time') {
            return renderTimePicker()
        }

        return renderDateTimePicker()
    }

    return (
        <MuiPickersUtilsProvider utils={DateFnsUtils} locale={locale}>
            {renderChildren()}
        </MuiPickersUtilsProvider>
    )
}

export default DateTime
