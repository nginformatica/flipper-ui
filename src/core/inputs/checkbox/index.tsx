import React from 'react'
import type { ChangeEvent, ReactNode } from 'react'
import MuiCheckbox from '@mui/material/Checkbox'
import MuiFormControlLabel from '@mui/material/FormControlLabel'
import MuiSwitch from '@mui/material/Switch'
import Typography from '@mui/material/Typography'
import type { DefaultProps } from '@/core/types'
import type { CheckboxProps } from '@mui/material/Checkbox'
import { HelperBox } from '@/core/inputs/text-field'
import { CheckFieldsWrapper, DENSE } from './styles'

export interface ICheckboxProps extends DefaultProps {
    name: string
    label?: ReactNode
    color?: 'primary' | 'secondary' | 'default'
    className?: string
    disabled?: boolean
    checked?: boolean
    dense?: boolean
    type?: 'switch' | 'checkbox'
    helperIcon?: ReactNode
    onHelperClick?: () => void
    onChange?: (event: ChangeEvent<HTMLElement>) => void
    checkboxProps?: CheckboxProps
}

const Checkbox = (props: ICheckboxProps) => {
    const {
        type = 'checkbox',
        margin,
        padding,
        style,
        helperIcon,
        onHelperClick,
        checkboxProps
    } = props

    const renderHelper = (
        <>
            {onHelperClick && (
                <HelperBox
                    helperIcon={helperIcon}
                    onHelperClick={onHelperClick}
                />
            )}
        </>
    )

    const renderCheckbox = (checkboxProps?: CheckboxProps) => (
        <MuiCheckbox
            {...checkboxProps}
            name={props.name}
            checked={props.checked}
            value={props.name}
            color={props.color}
            style={props.dense ? DENSE : {}}
            disabled={props.disabled}
            onChange={props.onChange}
        />
    )

    const renderSwitch = (checkboxProps?: Omit<CheckboxProps, 'size'>) => (
        <MuiSwitch
            {...checkboxProps}
            name={props.name}
            checked={props.checked}
            value={props.name}
            color={props.color}
            role='switch'
            disabled={props.disabled}
            onChange={props.onChange}
        />
    )

    const renderControl = (props?: CheckboxProps) =>
        type === 'checkbox' ? renderCheckbox(props) : renderSwitch(props)

    const renderLabel = () =>
        typeof props.label === 'string' && props.dense ? (
            <Typography variant='body2'>{props.label}</Typography>
        ) : (
            props.label
        )

    return (
        <CheckFieldsWrapper id={props.id}>
            {props.label ? (
                <MuiFormControlLabel
                    style={{ padding, margin, ...style }}
                    className={props.className}
                    label={renderLabel()}
                    control={renderControl(checkboxProps)}
                />
            ) : (
                renderControl(checkboxProps)
            )}
            {renderHelper}
        </CheckFieldsWrapper>
    )
}

export default Checkbox
