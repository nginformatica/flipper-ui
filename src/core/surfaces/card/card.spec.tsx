import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Button } from '@/core/inputs/button'
import {
    Card,
    CardActionArea,
    CardActions,
    CardContent,
    CardMedia,
    CardHeader
} from '.'

const url = 'valid_url'

describe('Card', () => {
    it('should render', () => {
        render(
            <Card name='test' data-testid='card'>
                <CardActionArea>
                    <CardMedia
                        image={url}
                        title='Puppies'
                        style={{ height: '256px' }}
                    />
                    <CardContent>Content</CardContent>
                </CardActionArea>
                <CardActions>
                    <Button size='small' color='primary'>
                        Confirm
                    </Button>
                </CardActions>
            </Card>
        )
        const card = screen.getByText('Content')

        expect(card).toBeDefined()
    })

    it('card header should spread props', () => {
        const title = 'This is a title'
        const subheader = 'This is a subheader'

        render(<CardHeader title={title} subheader={subheader} />)

        const cardHeader = screen.getByText(title)
        const cardSubheader = screen.getByText(subheader)

        expect(cardHeader).toBeDefined()
        expect(cardSubheader).toBeDefined()
    })
})
