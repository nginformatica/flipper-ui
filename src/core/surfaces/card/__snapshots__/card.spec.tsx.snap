// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root showable MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="card"
    name="test"
    role="mui-paper-container"
    style="padding: 24px; flex: 1; display: flex; flex-direction: column;"
  >
    <div
      class="sc-gEvDqW eZOuSv"
    >
      <h6
        class="MuiTypography-root sc-fqkwJk ktlWmc MuiTypography-h6 MuiTypography-colorPrimary"
        name="test-title"
      >
        title
      </h6>
      <button
        class="MuiButtonBase-root MuiIconButton-root sc-dcJtft kUpVZv"
        data-testid="remove-button"
        name="remove-test"
        style="margin: -10px 20px; padding: 0px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            color="#D84315"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root sc-dcJtft kUpVZv"
        data-testid="edit-button"
        name="cancel-test"
        style="margin: -10px; padding: 0px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      style="flex: 1; border: 1px solid #e0e0e0;"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined sc-aYaIB eiShaW MuiButton-outlinedPrimary"
      id="id-add-test"
      label="add button"
      name="add-test"
      style="margin: 12px 0px 24px; opacity: 1; border-style: dashed;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="NoteAddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm2 14h-3v3h-2v-3H8v-2h3v-3h2v3h3zm-3-7V3.5L18.5 9z"
          />
        </svg>
        add button
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <h1>
      test
    </h1>
  </div>
</div>
`;
